packages/twenty-server/src/engine/core-modules/event-emitter/utils/generate-fake-object-record-event.ts

```diff
@@ -0,0 +1,91 @@
+import { v4 } from 'uuid';
+
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { ObjectRecordCreateEvent } from 'src/engine/core-modules/event-emitter/types/object-record-create.event';
+import { generateFakeValue } from 'src/engine/utils/generate-fake-value';
+import { ObjectRecordUpdateEvent } from 'src/engine/core-modules/event-emitter/types/object-record-update.event';
+import { ObjectRecordDeleteEvent } from 'src/engine/core-modules/event-emitter/types/object-record-delete.event';
+import { ObjectRecordDestroyEvent } from 'src/engine/core-modules/event-emitter/types/object-record-destroy.event';
+
+export const generateFakeObjectRecordEvent = <Entity>(
+  objectMetadataEntity: ObjectMetadataEntity,
+  action: 'created' | 'updated' | 'deleted' | 'destroyed',
+):
+  | ObjectRecordCreateEvent<Entity>
+  | ObjectRecordUpdateEvent<Entity>
+  | ObjectRecordDeleteEvent<Entity>
+  | ObjectRecordDestroyEvent<Entity> => {
+  const recordId = v4();
+  const userId = v4();
+  const workspaceMemberId = v4();
+
+  const after = objectMetadataEntity.fields.reduce(
+    (acc, field) => {
+      acc[field.name] = generateFakeValue(field.type);
+
+      return acc;
+    },
+    {} as Record<string, any>,
+  );
+
+  if (action === 'created') {
+    return {
+      recordId,
+      userId,
+      workspaceMemberId,
+      objectMetadata: objectMetadataEntity,
+      properties: {
+        after: after as Entity,
+      },
+    };
+  }
```

Comment:
can you add `satisfies ObjectRecordCreateEvent<Entity>` (same for all retuned values). If we update the structure TS should break